ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/EvaluatorImpl.java

```diff
@@ -63,35 +63,26 @@ public void initialize() throws BallerinaShellException {
 
     public ShellCompilation getCompilation(String source) {
         PackageCompilation compilation;
+        ExceptionStatus exceptionStatus;
         try {
             Collection<Node> nodes = treeParser.parseString(source);
             Collection<Snippet> snippets = snippetFactory.createSnippets(nodes);
             compilation = invoker.getCompilation(snippets);
-            addAllDiagnostics(treeParser.diagnostics());
-            addAllDiagnostics(snippetFactory.diagnostics());
-            addAllDiagnostics(invoker.diagnostics());
-            treeParser.resetDiagnostics();
-            snippetFactory.resetDiagnostics();
-            invoker.resetDiagnostics();
-            return new ShellCompilation(Optional.ofNullable(compilation), ExceptionStatus.SUCCESS);
+            exceptionStatus = ExceptionStatus.SUCCESS;
+            clearDiagnostics(exceptionStatus);
+            return new ShellCompilation(compilation, exceptionStatus);
         } catch (TreeParserException e) {
-            addAllDiagnostics(treeParser.diagnostics());
-            treeParser.resetDiagnostics();
-            return new ShellCompilation(Optional.empty(), ExceptionStatus.TREE_PARSER_FAILED);
+            exceptionStatus = ExceptionStatus.TREE_PARSER_FAILED;
+            clearDiagnostics(exceptionStatus);
+            return new ShellCompilation(null, exceptionStatus);
```

Comment:
Without passing null, add a constructor with the status argument.

Reply:
Updated in 9f270db8b7259fb63e77545e8454623db4d6c3a7